tests/FSharp.Test.Utilities/TestConsole.fs

```diff
@@ -0,0 +1,89 @@
+namespace FSharp.Test
+
+open System
+open System.IO
+open System.Text
+open System.Threading
+
+module TestConsole =
+
+    /// Redirects reads performed on different async execution contexts to the relevant TextReader held by AsyncLocal.
+    type RedirectingTextReader() =
```

Comment:
Is it right to say that no one should be creating instances of the Reader/Writer types apart from this module?
(because a new instance would assign Null to the holder?)

If yes, can they or their constructors be private to the module?